controlplane/src/core/bufservices/organization/leaveOrganization.ts

```diff
@@ -49,7 +49,7 @@ export function leaveOrganization(
     }
 
     // the creator of the personal org cannot leave the organization.
-    if (org.creatorUserId === (authContext.userId || req.userID)) {
+    if (org.creatorUserId === (req.userID || authContext.userId)) {
```

Comment:
I have inverted this as the `authContext.userId` is always present so, if the `userId` is provided on the request, it was never being looked at